src/telemetry/logging.rs

```diff
@@ -1,39 +1,83 @@
-use std::env::current_dir;
-use std::path::Path;
-
-use eyre::Result;
-use tracing::subscriber::set_global_default;
-use tracing::{Level, Subscriber};
-use tracing_log::LogTracer;
-use tracing_subscriber::Layer;
-use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Registry};
+use std::{
+    env::current_dir,
+    path::{Path, PathBuf},
+};
+
+use tracing::Level;
+use tracing_appender::{
+    non_blocking::WorkerGuard,
+    rolling::{self, RollingFileAppender, Rotation},
+};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer};
 
 use ansi_term::Colour::{Blue, Cyan, Purple, Red, Yellow};
 
-/// Configure logging telemetry
-pub fn init(verbose: bool) -> Result<()> {
-    let subscriber = match verbose {
-        true => get_subscriber("magi=debug".into()),
-        false => get_subscriber("magi=info".into()),
-    };
-    init_subscriber(subscriber)
+/// Standard log file name prefix. This will be optionally appended with a timestamp
+/// depending on the rotation strategy.
+const LOG_FILE_NAME_PREFIX: &'static str = "magi.log";
+
+/// Configure logging telemetry with a global handler.
+pub fn init(
+    verbose: bool,
+    logs_dir: Option<String>,
+    logs_rotation: Option<String>,
+) -> Vec<WorkerGuard> {
+    // If a directory is provided, log to file and stdout
+    if let Some(dir) = logs_dir {
+        let directory = PathBuf::from(dir);
+        let rotation = get_rotation_strategy(&logs_rotation.unwrap_or("never".into()));
```

Comment:
Can we make the default rotation length daily? Just don't want to have unbounded disk usage by default.